### ISSUE TITLE
Integration tests for Playground CLI requests

### ISSUE THREAD

**Issue by lucifercr07** [DiceDB/dice#731]
## Description

DiceDB Playground is an interactive platform that allows users to experiment with DiceDB commands in a live environment, similar to the Go Playground. Users will be able to submit and execute commands directly through a web interface. The initial implementation will focus on enabling users to search for and trigger commands.  
For more details about DiceDB playground you can check [here](https://docs.google.com/document/d/1M5XfuOwQsRcl6g_VBr-FCLy71cnqCp7ulgyt6MsfESU/edit?usp=sharing).  

As part of this task we need to enhance unit tests and integration tests for `/cli` API.
This task would be extension of https://github.com/DiceDB/dice/issues/727

## Details

**Validation**:
- Validate the incoming request to ensure required parameters (`key`, `value`, `COMMAND_NAME`) are provided.
- Ensure appropriate HTTP response codes are returned based on success or failure scenarios.

**Testing**:
- Unit tests for the new `/cli` handler to cover success and failure scenarios.
- Integration tests that send various DiceDB commands via the CLI endpoint and check for correct responses.

**Comment by lucifercr07** [comment]
Dependent on https://github.com/DiceDB/dice/issues/727

**Comment by karandixit10** [comment]
Can I Take this up once it is implemented?

**Comment by lucifercr07** [comment]
@karandixit10 assigned, thanks for contributing. Please let me know if any other details required. Also you can collaborate with assignee of other issue.

**Comment by lucifercr07** [comment]
@karandixit10 please let us know if you require any help with this.

**Comment by karandixit10** [comment]
> @karandixit10 please let us know if you require any help with this.

I'll try to add this by EOD

**Comment by karandixit10** [comment]
Hi @lucifercr07  Do I have to create a mock DiceDB client to isolate the HTTP handler for Unit tests?

**Comment by karandixit10** [comment]
And will the folder structure for integration_tests be same as DiceDB tests?

**Comment by lucifercr07** [comment]
@karandixit10 we can keep the same.

**Comment by karandixit10** [comment]
Hi @lucifercr07 for integration test I have to run DiceDB server in the terminal so will it fail in the CI when pushed?

**Comment by arpitbbhayani** [comment]
Hello @karandixit10,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by piyushhhxyz** [comment]
@lucifercr07 @karandixit10 please assign me this issue, i'll be working on it

**Comment by yaten2302** [comment]
Hey @lucifercr07 @JyotinderSingh , I've a doubt regarding this issue, for writing the tests, those have to be written for [`playground-mono/internal/server/http.go`](https://github.com/DiceDB/playground-mono/blob/master/internal/server/http.go), am I correct?

**Comment by JyotinderSingh** [comment]
> Hey @lucifercr07 @JyotinderSingh , I've a doubt regarding this issue, for writing the tests, those have to be written for [`playground-mono/internal/server/http.go`](https://github.com/DiceDB/playground-mono/blob/master/internal/server/http.go), am I correct?

Yes, just ensure all edge cases and happy paths are being tested

**Comment by yaten2302** [comment]
Thanks! I'll create a PR for this soon 👍
